### ISSUE TITLE
Little Snitch 4 (warn mode) not compatible with OpenBazaar search

### ISSUE THREAD

**Issue by Kadrian** [OpenBazaar/openbazaar-desktop#1117]
**Brief Description**
OpenBazaar not usable together with Little Snitch 4. https://www.obdev.at/products/littlesnitch

Little Snitch in Warn mode monitors network traffic and allows you to allow / reject an outgoing connection to any IP you haven't been connected to before.

**Operating System**: OS X High Sierra 10.13.1
**OpenBazaar version**: 2.0.17
**Hardware**: Macbook Pro

**Reproducible**: Always

**Steps to reproduce**
1. Install OpenBazaar 2
2. Install and start Little Snitch 4 in "Warn" mode (asks about new connections)
3. Install and start OpenBazaar 2
4. Allow all (around 10) outgoing connections individually. That allows you to see your OpenBazaar profile (stable state until here)
5. Use the address bar in OpenBazaar to search for any term (hit enter)
6. See LittleSnitch using lots of CPU and blocking OpenBazaar from continuing. This state goes on (after all I could observe) forever.

**Observed Behavior**
Little Snitch uses plenty of CPU and stops OpenBazaar from working. The only way around it is to stop the network filter in little snitch (basically let everything through), quit OpenBazaar, re-start the network filter.

I suspect this is due to the decentralized nature of OpenBazaar during search, when it opens lots of outgoing connections to lots of different IP addresses.

I'm curious: Is that what's happening – and is that the desired behavior? How many connections are opened at the same time?

**Expected Behavior**
Little Snitch continues to work and asks me to allow one connection after the other.

**Comment by rmisio** [comment]
This is more of a question for the [server repo](https://github.com/OpenBazaar/openbazaar-go). I think the culprit is the server using IPFS which is known to open a lot of concurrent connections (in fact, it's so many that often times operations fail on mac due to too many open connections - mac has a quite low limit by default).

**Comment by Kadrian** [comment]
Okay, what does it mean that you close the issue? Not a resolution possible on your side? It's not relevant?

**Comment by rmisio** [comment]
> Okay, what does it mean that you close the issue? Not a resolution possible on your side?

Yes, I don't think there's a resolution on our side (or the server's really). It's IPFS that is opening up many connections. We've been in talks with them to see about reducing that number, but I'm not sure it will ever be below Little Snitches threshold. I think an inherent part of the architecture of IPFS requires that they have many concurrent open connections.

**Comment by hoffmabc** [comment]
I am taking a look at this @rmisio. We can leave it open while I analyze it and then I’ll close it if necessary.

**Comment by jjeffryes** [comment]
@hoffmabc you were going to take a look at this, did you get a chance to do that?
